src/ddscxx/include/dds/pub/detail/DataWriter.hpp

```diff
@@ -69,6 +69,12 @@ class dds::pub::detail::DataWriter : public ::org::eclipse::cyclonedds::pub::Any
 
     void init(ObjectDelegate::weak_ref_type weak_ref);
 
+    bool is_loan_supported();
+
+    T& loan_sample();
```

Comment:
Doesn't this overlap with the same changes on PR#125?

Reply:
@reicheratwork PR#125 has this commit 0ee6ff20b50d02a2bab8168f0ba3060b62200de4 for adding return loan functionality, which I already incorporated in this MR